src/nodedumper.cc

```diff
@@ -8,6 +8,46 @@
 #include <assert.h>
 #include <boost/regex.hpp>
 
+
+void GroupNodeChecker::incChildCount(int groupNodeIndex) {
+	auto search = this->groupChildCounts.find(groupNodeIndex);
+	// if no entry then given node wasn't a group node
+	if (search != this->groupChildCounts.end()) {
+		++(search->second);
+	}
+}
+
+int GroupNodeChecker::getChildCount(int groupNodeIndex) {
+	auto search = this->groupChildCounts.find(groupNodeIndex);
+	if (search != this->groupChildCounts.end()) {
+		return search->second;
+	} else {
+		return -1;
+	}
+}
+
+Response GroupNodeChecker::visit(State &state, const GroupNode &node)
+{
+	if (state.isPrefix()) {
+		// create entry for group node, which children may increment
+		this->groupChildCounts.emplace(std::make_pair(node.index(),0));
```

Comment:
emplace() doesn't actually want a pair, but the pair constructor arguments directly, so if you do this, you save a copy:
```
		this->groupChildCounts.emplace(node.index(), 0);
```